### PR TITLE
ch-109_geotargeting

### PR DESCRIPTION
**Pull request by StareIntoTheBeard**

### CODE REVIEW

File: lib/code_fund_web/controllers/ad_serve_controller.ex [hunk 1]
```diff
@@ -86,6 +87,11 @@ defmodule CodeFundWeb.AdServeController do
         )
         |> details_render(conn)
 
+      true ->
+        conn
+        |> error_details(property_id, "Your country cannot view sponsorships at this time")
```
Comment (coderberry) [inline lib/code_fund_web/controllers/ad_serve_controller.ex]: Let's replace this text with "CodeFund does not have an advertiser for you at this time"

Comment (StareIntoTheBeard) [reply lib/code_fund_web/controllers/ad_serve_controller.ex]: sounds good, i figured that would change

File: config/configs/geolix.exs [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+config :geolix,
+  databases: [
+    %{
+      id: :country,
+      adapter: Geolix.Adapter.MMDB2,
+      init: {MaxMindInitializer, :init}
+    }
+  ],
+  banned_countries: System.get_env("BANNED_COUNTRIES"),
+  database_location:  System.get_env("DATABASE_LOCATION")
```
Comment (coderberry) [inline config/configs/geolix.exs]: Just a comment.. maybe we should qualify this env var for maxmind.. e.g. MMDB_LOCATION ?

File: config/configs/geolix.exs [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+config :geolix,
+  databases: [
+    %{
+      id: :country,
+      adapter: Geolix.Adapter.MMDB2,
+      init: {MaxMindInitializer, :init}
+    }
+  ],
+  banned_countries: System.get_env("BANNED_COUNTRIES"),
+  database_location: System.get_env("DATABASE_LOCATION")
```
Comment (coderberry) [inline config/configs/geolix.exs]: Just a suggestion.. maybe we should use something like `MMDB_LOCATION` to make it so devs don't think it's location for Postgres

File: lib/framework/geolocation.ex [hunk 4]
```diff
@@ -0,0 +1,17 @@
+defmodule Framework.Geolocation do
+  @spec is_banned_country?({integer, integer, integer, integer}) :: boolean
```
Comment (coderberry) [inline lib/framework/geolocation.ex]: Do you know if this will work if the IP coming in is an IPv6?

File: config/configs/geolix.exs (hunk 3 above)
Comment (StareIntoTheBeard) [reply config/configs/geolix.exs]: good call

**Pull request merged**